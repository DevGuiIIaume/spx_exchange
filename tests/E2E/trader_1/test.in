1
[T0] BUY 0 GPU 10 500;
[T0] SELL 1 Router 20 600;
[T0] SELL 2 GPU 1000 100;
[T0] DISCONNECT;