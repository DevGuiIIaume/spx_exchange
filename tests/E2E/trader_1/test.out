[SPX] Starting
[SPX] Trading 2 products: GPU Router
[SPX] Created FIFO /tmp/spx_exchange_0
[SPX] Created FIFO /tmp/spx_trader_0
[SPX] Connected to /tmp/spx_exchange_0
[SPX] Connected to /tmp/spx_trader_0
[SPX] Created FIFO /tmp/spx_exchange_1
[SPX] Created FIFO /tmp/spx_trader_1
[SPX] Connected to /tmp/spx_exchange_1
[SPX] Connected to /tmp/spx_trader_1
[SPX] [T0] Parsing command: <BUY 0 GPU 10 500>
[SPX]	--ORDERBOOK--
[SPX]	Product: GPU; Buy levels: 1; Sell levels: 0
[SPX]		BUY 10 @ $500 (1 order)
[SPX]	Product: Router; Buy levels: 0; Sell levels: 0
[SPX]	--POSITIONS--
[SPX]	Trader 0: GPU 0 ($0), Router 0 ($0)
[SPX]	Trader 1: GPU 0 ($0), Router 0 ($0)
[SPX] [T0] Parsing command: <SELL 1 Router 20 600>
[SPX]	--ORDERBOOK--
[SPX]	Product: GPU; Buy levels: 1; Sell levels: 0
[SPX]		BUY 10 @ $500 (1 order)
[SPX]	Product: Router; Buy levels: 0; Sell levels: 1
[SPX]		SELL 20 @ $600 (1 order)
[SPX]	--POSITIONS--
[SPX]	Trader 0: GPU 0 ($0), Router 0 ($0)
[SPX]	Trader 1: GPU 0 ($0), Router 0 ($0)
[SPX] [T1] Parsing command: <BUY 0 Router 20 600>
[SPX] Match: Order 1 [T0], New Order 0 [T1], value: $12000, fee: $120.
[SPX]	--ORDERBOOK--
[SPX]	Product: GPU; Buy levels: 1; Sell levels: 0
[SPX]		BUY 10 @ $500 (1 order)
[SPX]	Product: Router; Buy levels: 0; Sell levels: 0
[SPX]	--POSITIONS--
[SPX]	Trader 0: GPU 0 ($0), Router -20 ($12000)
[SPX]	Trader 1: GPU 0 ($0), Router 20 ($-12120)
[SPX] [T0] Parsing command: <SELL 2 GPU 1000 100>
[SPX] Match: Order 0 [T0], New Order 2 [T0], value: $5000, fee: $50.
[SPX]	--ORDERBOOK--
[SPX]	Product: GPU; Buy levels: 0; Sell levels: 1
[SPX]		SELL 990 @ $100 (1 order)
[SPX]	Product: Router; Buy levels: 0; Sell levels: 0
[SPX]	--POSITIONS--
[SPX]	Trader 0: GPU 0 ($-50), Router -20 ($12000)
[SPX]	Trader 1: GPU 0 ($0), Router 20 ($-12120)
[SPX] Trader 1 disconnected
[SPX] Trader 0 disconnected
[SPX] Trading completed
[SPX] Exchange fees collected: $170
